### ISSUE TITLE
Lagging desktop when using entertainment area sync

### ISSUE THREAD

**Issue by Ex0danify** [vchlum/hue-lights#44]
Hey there,

When I start using the LightSync for entertainment areas on my Linux desktop, the whole system gets very choppy/laggy.

MySetup:
- CPU: Ryzen Threadripper 2950x
- GPU: Nvidia RTX2070
- RAM: 32GB DDR4@3200MHz
- OS: Pop_OS 21.10
- Kernel: 5.16.7
- GnomeShell: 40.4.0
- Window Server: X11
- Resolution: 3440x1440@120Hz, DisplayPort
- Nvidia Driver: 510.47.03

Here is the [Debug Log](https://gist.github.com/Ex0danify/9f49bdaaabe7496645169cab13ccc73f).
For the log I started LightSync, moved some windows around and the stopped it again. If you need further information, please tell me.

Thanks for your great contribution

Cheers
Ex0

**Comment by vchlum** [comment]
Hi @Ex0danify How many lights are in your entertainment group? Are there any gradient lights? In my experience, 10 lights (including one gradient light strip) work fine on the less powerful computer...

- Could you create&test an entertainment group with only one light?
- Could you test the mode 'random' on your entertainment group? Is it laggy too?

**Comment by Ex0danify** [comment]
Thanks a lot man, highly appreciated!

**Comment by vchlum** [comment]
In my virtual box instance of pop_os, the issue is not recognizable. Could you please run the `top` command and check the difference with and without syncing? Please, try to identify what is utilizing your machine.

**Comment by vchlum** [comment]
This is actually expected. gnome-shell is the right process. I was curious if there is anything else related.

My gnome-shell process keeps it around 20-25%, and it goes to 10% sometimes. My machine is i7, intel xe, 16GB RAM. Definitely less powerful.

Could you please try to decrease the intensity of synchronization to the lowest value? It should decrease the utilization. I checked again on my bare metal and with high intensity, the lagging is noticeable too. Usually, I use middle intensity.

I will definitely look into optimization, but it uses a gnome-based feature to get color from the screen. I am not sure how to optimize it yet.

If decreasing intensity will relieve your issue, I can change the scale of intensity to allow a lower value. This would be one solution. I believe.

**Comment by Ex0danify** [comment]
Indeed, lowering the sensitivity makes the lag less noticeable, though it doesn't vanish completely.

**Comment by vchlum** [comment]
OK, you can set the default intensity to a lower value in the settings. I will try to optimize it, or I will decrease the lowest possible intensity.
